### PR TITLE
UIAmountInput with new architecture.

### PR DESCRIPTION
**Pull request by SAnatoliiS**

### CODE REVIEW

File: kit/inputs/src/MaterialTextView/hooks/useApplyMask/amount/runUIFormat.ts [hunk 1]
```diff
@@ -61,7 +61,7 @@ export function runUIFormat(
             normalizedDecimalPart = normalizedDecimalPart.slice(0, countOfDecimalDigits);
         }
 
-        normalizedText += delimeter;
+        normalizedText += '.';
```
Comment (sertony) [inline kit/inputs/src/MaterialTextView/hooks/useApplyMask/amount/runUIFormat.ts]: I don't think that it should be a dot symbol here (as the delimiter depends on the localization)

File: kit/inputs/src/UIAmountInput/UIAmountInputContent.tsx [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+            onHover,
+            onSelectionChange,
+            onChangeAmount: onChangeAmountProp,
+            decimalAspect = UIAmountInputDecimalAspect.Precision,
+            multiline,
+            message,
+            messageType = UIAmountInputMessageType.Info,
+            label,
+        } = props;
+        // @ts-ignore
```
Comment (sertony) [inline kit/inputs/src/UIAmountInput/UIAmountInputContent.tsx:57]: What's the problem?

File: kit/inputs/src/UIAmountInput/hooks/amountMask/runUIFormat.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        decimalPart != null &&
+        decimalPart.length > 0
+    ) {
+        let normalizedDecimalPart = decimalPart.replace(notNumbersRegexp, '');
+
+        if (countOfDecimalDigits !== null) {
+            normalizedDecimalPart = normalizedDecimalPart.slice(0, countOfDecimalDigits);
+        }
+
+        normalizedText += '.';
```
Comment (sertony) [inline kit/inputs/src/UIAmountInput/hooks/amountMask/runUIFormat.ts]: What about the dot again?

File: kit/inputs/src/UIAmountInput/hooks/useSetText.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+const defaultConfig: FormatAndSetTextConfig = {
+    shouldSetText: true,
+    callOnChangeProp: true,
+};
+export function useSetText(
+    ref: React.RefObject<UITextViewRef>,
+    onChangeAmountProp: UIAmountInputProps['onChangeAmount'],
+): SetText {
+    // @ts-expect-error
```
Comment (sertony) [inline kit/inputs/src/UIAmountInput/hooks/useSetText.ts:17]: Also, what's the problem?

File: kit/inputs/src/UITextView/UITextView.tsx [hunk 5]
```diff
@@ -10,15 +10,29 @@ const inputTypographyStyle = StyleSheet.flatten(Typography[textViewTypographyVar
...
+    const { fontSize, lineHeight } = inputTypographyStyle;
+    if (fontSize == null || lineHeight == null) {
+        return {};
+    }
+    let textInputHeight = 0;
+    /**
+     * This calculation were carried out experimentally.
+     */
+    if (fontSize <= 20) {
+        textInputHeight = 27.636363983154297;
```
Comment (sertony) [inline kit/inputs/src/UITextView/UITextView.tsx:26]: 😨

File: kit/inputs/src/UITextView/UITextView.tsx [hunk 6]
```diff
@@ -10,15 +10,29 @@ const inputTypographyStyle = StyleSheet.flatten(Typography[textViewTypographyVar
 const textViewLineHeight = inputTypographyStyle.lineHeight ?? UILayoutConstant.smallCellHeight;
 
 /**
- * The singleline input on the ios platform has a bug -
- * if `(lineHeight - fontSize) > 5` it starts to crop
- * letters from the bottom (letters like "g", "j").
- * Style prop `textAlignVertical` works only on Android.
+ * Android TextInput does not respond to lineHeight changes,
```
Comment (sertony) [inline kit/inputs/src/UITextView/UITextView.tsx:13]: This is probably not working well on all Android devices...

File: kit/inputs/src/MaterialTextView/hooks/useApplyMask/amount/runUIFormat.ts (hunk 1 above)
Comment (SAnatoliiS) [reply kit/inputs/src/MaterialTextView/hooks/useApplyMask/amount/runUIFormat.ts]: This has been changed for testing purposes. I just forgot to undo these changes.

File: kit/inputs/src/UIAmountInput/UIAmountInputContent.tsx (hunk 2 above)
Comment (SAnatoliiS) [reply kit/inputs/src/UIAmountInput/UIAmountInputContent.tsx:57]: The types don't match with `useAnimatedRef`. For this purposes it doesn't matter.

File: kit/inputs/src/UIAmountInput/hooks/amountMask/runUIFormat.ts (hunk 3 above)
Comment (SAnatoliiS) [reply kit/inputs/src/UIAmountInput/hooks/amountMask/runUIFormat.ts]: Replaced by `delimeter`.

File: kit/inputs/src/UIAmountInput/hooks/useSetText.ts (hunk 4 above)
Comment (SAnatoliiS) [reply kit/inputs/src/UIAmountInput/hooks/useSetText.ts:17]: Also problem with the types don't match with `useAnimatedRef`

File: kit/inputs/src/UITextView/UITextView.tsx (hunk 6 above)
Comment (SAnatoliiS) [reply kit/inputs/src/UITextView/UITextView.tsx:13]: I didn't notice anything wrong. I think we should test it, and fix problems only if they found.

File: kit/inputs/src/UITextView/UITextView.tsx (hunk 5 above)
Comment (SAnatoliiS) [reply kit/inputs/src/UITextView/UITextView.tsx:26]: This was measured. Logic of android native TextInput size is not obvious.

**Review by sertony — APPROVED** [review]

**Pull request merged**